subprojects/jacoco/src/main/java/org/gradle/testing/jacoco/tasks/JacocoMerge.java

```diff
@@ -62,11 +63,11 @@ public void setExecutionData(FileCollection executionData) {
      */
     @OutputFile
     public File getDestinationFile() {
-        return destinationFile;
+        return destinationFile.get();
     }
 
     public void setDestinationFile(File destinationFile) {
-        this.destinationFile = destinationFile;
+        this.destinationFile.set(destinationFile);
     }
```

Comment:
should be able to either get or set the property as a `Provider<File>`, otherwise there's not really any point making the field a `PropertySet<File>`.